crates/bin/c8y-device-management/src/main.rs

```diff
@@ -77,3 +89,12 @@ async fn main() -> anyhow::Result<()> {
     runtime.run_to_completion().await?;
     Ok(())
 }
+
+fn mqtt_config(tedge_config: &TEdgeConfig) -> Result<MqttConfig, TEdgeConfigError> {
```

Comment:
Instead of this function, I'd have preferred something like `TryFrom<&TEdgeConfig> for MqttConifg` but the compiler was not letting me do that with the error: `error[E0117]: only traits defined in the current crate can be implemented for types defined outside of the crate`.

The other option would have been either:
* Introduce a concrete wrapper impl for `MqttConifg` and implement the `TryFrom` for that type
* Implement the `TryFrom` for the `mqtt_channel::Config` itself. But, for that I need to make the `mqtt_channel` crate depend on the `tedge_config` crate as well, which I was not sure was a good idea.

Since both were additional work for little benefit, I chose this path. But happy to fall back to either approach if you prefer one of those as well.

Reply:
I'm okay with that.

A 3rd alternative is to `TryFrom<&TEdgeConfig> for MqttConfig` in the `tedge_config` crate. This would add many dependencies to  `tedge_config` but I see those less problematic.